### PR TITLE
Add endpoint to fetch Bitcoin price via Tor

### PR DESCRIPTION
**Pull request by louneskmt**
New constants: environment variables for Tor proxy
Adds package `socks-proxy-agent` to route request via Tor
Adds a new endpoint `/utils/price` fetching Bitcoin price with [CryptoCompare](cryptocompare.com) API, via Tor

### CODE REVIEW

File: utils/const.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ module.exports = {
   UPDATE_SIGNAL_FILE: process.env.UPDATE_SIGNAL_FILE || '/signals/update',
   UPDATE_LOCK_FILE: process.env.UPDATE_LOCK_FILE || '/statuses/update-in-progress',
   BACKUP_STATUS_FILE: process.env.BACKUP_STATUS_FILE || '/statuses/backup-status.json',
+  TOR_PROXY_IP: process.env.TOR_PROXY_IP,
+  TOR_PROXY_PORT: process.env.TOR_PROXY_PORT,
```
Comment (lukechilds) [inline utils/const.js:33]: Maybe set some sensible defaults for these values?

I'd suggest 192.168.0.1 and 9050, the default when Tor is running on the same machine.

They should also be documented in the table in the readme: https://github.com/getumbrel/umbrel-manager#step-2-set-environment-variables

File: routes/v1/utils.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const auth = require('middlewares/auth.js');
+
+const constants = require('utils/const.js');
+const safeHandler = require('utils/safeHandler');
+
+const {SocksProxyAgent} = require('socks-proxy-agent');
+const axios = require('axios');
+
+const agent = new SocksProxyAgent(`socks5h://${constants.TOR_PROXY_IP}:${constants.TOR_PROXY_PORT}`);
```
Comment (lukechilds) [inline routes/v1/utils.js:12]: What does the `h` mean in `socks5h` here?

Does it still work ok if you remove it?

File: app.js [hunk 3]
```diff
@@ -40,6 +41,7 @@ app.use(morgan(logger.morganConfiguration));
 app.use('/ping', ping);
 app.use('/v1/account', account);
 app.use('/v1/system', system);
+app.use('/v1/utils', utils);
```
Comment (lukechilds) [inline app.js:44]: I'm not too keen on the name `utils` for the route, I think it should be something more descriptive. Maybe `external` and have this as a place to proxy through any other external API requests in the future. Or alternatively just `price`?

**Review by lukechilds — CHANGES_REQUESTED** [review]
Looks good!

Couple of minor nits but other than that, great stuff.

**Comment by lukechilds** [comment]
Pro-tip @louneskmt.

If you add your Docker Hub credentials as secrets to GitHub, the action in your fork repo will auto deply images to Docker Hub for us to easily pull down and test.

https://github.com/louneskmt/umbrel-manager/blob/master/.github/workflows/on-push.yml

`DOCKER_PASSWORD` = API token you generate on Docker Hub
`DOCKER_USERNAME` = Your Docker Hub username
`DOCKER_HUB_USER` = Your Docker Hub username

And then on next push there will be a public image available at `$DOCKER_HUB_USER/manager:$BRANCH_NAME`.

Same for your dashboard fork too.

File: routes/v1/utils.js (hunk 2 above)
Comment (louneskmt) [reply routes/v1/utils.js:12]: Based on [this part](https://github.com/TooTallNate/node-socks-proxy-agent/blob/8ed8efa99c8a17a8ee52872728f13a94497ddc13/src/agent.ts#L51) of package source, the `5h` corresponds to the socks version (as it's possible to use other versions like `4`, `4a`, or `5`). If no version is specified, the default is set to `5h`. So deleting it won't do anything, but the socks version is clear. As you like.

File: app.js (hunk 3 above)
Comment (louneskmt) [reply app.js:44]:
```suggestion
app.use('/v1/external', external);
```
Seems go to me!

**Comment by louneskmt** [comment]
> Pro-tip @louneskmt.
> 
> If you add your Docker Hub credentials as secrets to GitHub, the action in your fork repo will auto deploy images to Docker Hub for us to easily pull down and test.
> 
> https://github.com/louneskmt/umbrel-manager/blob/master/.github/workflows/on-push.yml
> 
> `DOCKER_PASSWORD` = API token you generate on Docker Hub
> `DOCKER_USERNAME` = Your Docker Hub username
> `DOCKER_HUB_USER` = Your Docker Hub username
> 
> And then on next push there will be a public image available at `$DOCKER_HUB_USER/manager:$BRANCH_NAME`.
> 
> Same for your dashboard fork too.

Thanks!

File: routes/v1/utils.js (hunk 2 above)
Comment (lukechilds) [reply routes/v1/utils.js:12]: According to [this](https://github.com/urllib3/urllib3/issues/1035#issue-188967574) `socks5h` means the hostname is resolved by the SOCKS server and `socks5` means it's resolved locally.

So yeah, we definitely want `socks5h`!

**Review by lukechilds — APPROVED** [review]
Perfect!

**Pull request merged**